corehq/apps/userreports/tasks.py

```diff
@@ -178,25 +186,52 @@ def _run_report(backend_to_use):
     return objects
 
 
+@periodic_task(
+    run_every=ASYNC_INDICATOR_QUEUE_TIME,
+    queue=settings.CELERY_PERIODIC_QUEUE,
+)
+def queue_async_indicators():
+    start = datetime.utcnow()
+    cutoff = start + ASYNC_INDICATOR_QUEUE_TIME
+    time_for_crit_section = ASYNC_INDICATOR_QUEUE_TIME.seconds - 10
+
+    with CriticalSection(['queue-async-indicators'], timeout=time_for_crit_section):
+        redis_client = get_redis_client().client.get_client()
+        indicators = AsyncIndicator.objects.all()[:10000]
```

Comment:
There's a lock for 6 hours on queueing the same indicator and it's only possible to queue 10k indicators at one time because this will only pull the oldest created indicators thanks to https://github.com/dimagi/commcare-hq/pull/15462/files#diff-3af365e79b93f051d416585bcba61b7eR703

Maybe we should bump up the 6 hours? 6 hours assumes about 1 indicator every 2 seconds with one thread. if we have 100 concurrency (same as repeaters on prod) then each indicator could take 200 seconds before we run into issues.

We also have proactive warnings about too many tasks in the celery queue so I don't think it's something that will slide by unnoticed (assuming datadog can be put on nic eventually)